ws-security-dom/src/main/java/org/apache/wss4j/dom/message/WSSecEncryptedKey.java

```diff
@@ -277,127 +277,144 @@ protected void createEncryptedKeyElement(X509Certificate remoteCert, Crypto cryp
         if (customEKKeyInfoElement != null) {
             encryptedKeyElement.appendChild(getDocument().adoptNode(customEKKeyInfoElement));
         } else {
+            Element keyInfoElement =
+                    getDocument().createElementNS(
+                            WSConstants.SIG_NS, WSConstants.SIG_PREFIX + ":" + WSConstants.KEYINFO_LN
+                    );
+            keyInfoElement.setAttributeNS(
+                    WSConstants.XMLNS_NS, "xmlns:" + WSConstants.SIG_PREFIX, WSConstants.SIG_NS
+            );
+
+            Element keyInfoChildElement;
+
```

Comment:
I think I'd prefer the "else" clause to stay the same as before, as you're mixing the SecurityTokenReference case with this new case of not having one. Instead please consider putting an "else if" clause before it for WSConstants.X509_SKI